packages/valory/skills/trader_abci/composition.py

```diff
@@ -88,9 +95,12 @@
     FailedTransactionSubmissionRound: HandleFailedTxRound,
     FinishedDecisionRequestTxRound: DecisionReceiveRound,
     FinishedBetPlacementTxRound: RedeemRound,
-    FinishedRedeemingTxRound: ResetAndPauseRound,
+    FinishedRedeemingTxRound: CallCheckpointRound,
```

Comment:
This means all services, even non-staking ones need to transition here...

Reply:
Correct. Conceptually, it would not make much sense to add a check to any of the existing rounds on whether we should transition here or not. Therefore, `CallCheckpointRound` is always entered and immediately exits if the service is not staked.